### ISSUE TITLE
ASM: 'azure vm endpoint acl-rule create' throws 'Cannot read property toLowerCase of undefined'

### ISSUE THREAD

**Issue by romangromov** [Azure/azure-xplat-cli#2851]
CLI Version: **0.9.20**
OS Type: Win 
Installation via: any

Mode: **ASM**

Description:
`azure vm show` command doesn't output VM name.

Steps to reproduce:
1) Trying create ACL rule for existinv VM and endpoint attached:

```
azure vm endpoint acl-rule create tryaVm TestEndpoint 10 --remote-subnet "10.0.0.0/32"

info:    Executing command vm endpoint acl-rule create
error:   Cannot read property 'toLowerCase' of undefined
info:    Error information has been recorded to C:\Users\roman.gromov\.azure\azure.err
error:   vm endpoint acl-rule create command failed
```

Stack:
```
2016-05-04T09:12:32.743Z:
[Error: An ACL rule with order 12 not found for this endpoint]
Error: An ACL rule with order 12 not found for this endpoint
    at D:\Projects\Nodejs\azure-xplat-cli-ms\lib\commands\asm\vm\vmclient.js:2109:33
    at D:\Projects\Nodejs\azure-xplat-cli-ms\node_modules\azure-asm-compute\lib\computeManagementClient.js:28498:14
    at D:\Projects\Nodejs\azure-xplat-cli-ms\lib\util\logging.js:412:7
    at handleRedirect (D:\Projects\Nodejs\azure-xplat-cli-ms\lib\util\utils.js:520:9)
    at D:\Projects\Nodejs\azure-xplat-cli-ms\lib\util\utils.js:543:7
    at handleRedirect (D:\Projects\Nodejs\azure-xplat-cli-ms\node_modules\azure-common\lib\services\filters\redirectfilter.js:36:9)
    at D:\Projects\Nodejs\azure-xplat-cli-ms\node_modules\azure-common\lib\services\filters\errorhandlingfilter.js:49:7
    at Request._callback (D:\Projects\Nodejs\azure-xplat-cli-ms\node_modules\azure-common\lib\http\request-pipeline.js:109:14)
    at Request.self.callback (D:\Projects\Nodejs\azure-xplat-cli-ms\node_modules\azure-common\node_modules\request\request.js:199:22)
    at emitTwo (events.js:87:13)

```

**Comment by eduardkoller** [comment]
@huangpf - can you please take a look?

**Comment by MikhailTryakhov** [comment]
fixed by [PR#2969](https://github.com/Azure/azure-xplat-cli/pull/2969)
